examples/lighting-app/lighting-common/BUILD.gn

```diff
@@ -36,41 +37,13 @@ if (enable_pw_rpc) {
   }
 }
 
-source_set("lighting-common") {
-  sources = [
-    "${chip_root}/src/app/clusters/bindings/bindings.cpp",
-    "${chip_root}/src/app/clusters/level-control/level-control.cpp",
-    "${chip_root}/src/app/clusters/network-commissioning/network-commissioning-ember.cpp",
-    "${chip_root}/src/app/clusters/network-commissioning/network-commissioning.cpp",
-    "${chip_root}/src/app/clusters/on-off-server/on-off.cpp",
-    "${chip_root}/src/app/reporting/reporting-default-configuration.cpp",
-    "${chip_root}/src/app/reporting/reporting.cpp",
-    "${chip_root}/src/app/server/DataModelHandler.cpp",
-    "${chip_root}/src/app/util/af-event.cpp",
-    "${chip_root}/src/app/util/af-main-common.cpp",
-    "${chip_root}/src/app/util/attribute-size.cpp",
-    "${chip_root}/src/app/util/attribute-storage.cpp",
-    "${chip_root}/src/app/util/attribute-table.cpp",
-    "${chip_root}/src/app/util/binding-table.cpp",
-    "${chip_root}/src/app/util/chip-message-send.cpp",
-    "${chip_root}/src/app/util/client-api.cpp",
-    "${chip_root}/src/app/util/ember-print.cpp",
-    "${chip_root}/src/app/util/message.cpp",
-    "${chip_root}/src/app/util/process-cluster-message.cpp",
-    "${chip_root}/src/app/util/process-global-message.cpp",
-    "${chip_root}/src/app/util/util.cpp",
-    "gen/call-command-handler.cpp",
-    "gen/callback-stub.cpp",
+chip_data_model("lighting-common") {
+  cluster_sources = [
```

Comment:
lighting and lock apps should also include `basic` sources even though they're currently unused (ZAP files of the apps contain the Basic cluster).